jni/src/nmslib_wrapper.cpp

```diff
@@ -174,17 +172,25 @@ void knn_jni::nmslib_wrapper::CreateIndex(knn_jni::JNIUtilInterface *jniUtil, JN
                                                                                   *(space),
                                                                                   dataset));
     index->CreateIndex(similarity::AnyParams(indexParameters));
-    index->SaveIndex(indexPathCpp);
 
-    for (auto &it : dataset) {
+    knn_jni::stream::NativeEngineIndexOutputMediator mediator {jniUtil, env, output};
+    knn_jni::stream::NmslibOpenSearchIOWriter writer {&mediator};
+
+    if (auto hnswFloatIndex = dynamic_cast<similarity::Hnsw<float> *>(index.get())) {
+      hnswFloatIndex->SaveIndexWithStream(writer);
+      mediator.flush();
+    } else {
```

Comment:
why flush function cannot be combined with `SaveIndexWithStream` function?

Reply:
There's no reason to do it!
I will update NMSLIB patch to do it inside in the end before returning.